### PR TITLE
fix(curriculum): external comma external link in CIP

### PR DESCRIPTION
**Pull request by Sboonny**
Checklist:

- [x] I have read [freeCodeCamp's contribution guidelines](https://contribute.freecodecamp.org).
- [x] My pull request has a descriptive title (not a vague title like `Update index.md`)
- [x] My pull request targets the `main` branch of freeCodeCamp.
- [x] I have tested these changes either locally on my machine, or GitPod.

127 in #40749 

Sorry Jeremy, I should have added waiting update label, or took minute to update the challenge

### CODE REVIEW

File: curriculum/challenges/english/10-coding-interview-prep/rosetta-code/comma-quibbling.md [hunk 1]
```diff
@@ -52,19 +41,19 @@ assert(typeof quibble(['ABC']) === 'string');
...
 
-`quibble(["ABC", "DEF"])` should return "{ABC and DEF}".
+`quibble(["freeCodeCamp", "Curriculum"])` should return "{freeCodeCamp and Curriculum}".
 
 ```js
 assert.equal(quibble(testCases[2]), results[2]);
 ```
 
-`quibble(["ABC", "DEF", "G", "H"])` should return "{ABC,DEF,G and H}".
+`quibble(["freeCodeCamp", "Curriculum", "Articles", "Volunteers"])` should return "{freeCodeCamp,Curriculum,Articles and Volunteers}".
```
Comment (jeremylt) [inline curriculum/challenges/english/10-coding-interview-prep/rosetta-code/comma-quibbling.md:56]: Do we really want this challenge to omit spaces? That's not consistent with the example above
```suggestion
`quibble(["freeCodeCamp", "Curriculum", "Articles", "Volunteers"])` should return "{freeCodeCamp, Curriculum, Articles and Volunteers}".
```

**Review by ShaunSHamilton — DISMISSED** [review]
Ditto what Jeremy said in some commit - the solution should be invalid

**Review by Sembauke — DISMISSED** [review]

File: curriculum/challenges/english/10-coding-interview-prep/rosetta-code/comma-quibbling.md [hunk 2]
```diff
@@ -8,29 +8,18 @@ dashedName: comma-quibbling
 
 # --description--
 
-Comma quibbling is a task originally set by Eric Lippert in his [blog](https://blogs.msdn.com/b/ericlippert/archive/2009/04/15/comma-quibbling.aspx).
+Comma quibbling is a task originally set by Eric Lippert in his 2009 blog. In this challenge, you will create a `string` from an `array`. You have to account for the `array` having none items, single item, or multiple items in it.
```
Comment (moT01) [inline curriculum/challenges/english/10-coding-interview-prep/rosetta-code/comma-quibbling.md:11]:
```suggestion
Comma quibbling is a task originally set by Eric Lippert in his 2009 blog. In this challenge, you will create a `string` from an `array`. You have to account for the `array` having no items, a single item, or multiple items in it. 
```

**Review by moT01 — CHANGES_REQUESTED** [review]
I don't understand why we are changing the instructions here so significantly. It is straying from the original rosetta challenge quite a bit.

File: curriculum/challenges/english/10-coding-interview-prep/rosetta-code/comma-quibbling.md (hunk 2 above)
Comment (Sboonny) [reply curriculum/challenges/english/10-coding-interview-prep/rosetta-code/comma-quibbling.md:11]: I will look back to the old one and try to implement it back

**Comment by Sboonny** [comment]
> I don't understand why we are changing the instructions here so significantly. It is straying from the original rosetta challenge quite a bit.

@moT01 Oh, I changed it, because they were similar to this https://rosettacode.org/wiki/Comma_quibbling. 

But this doesn't change that I missed the original intent, if it's possible to elaborate more, because I am still missing it.😅

**Comment by moT01** [comment]
> I changed it because they were similar to this https://rosettacode.org/wiki/Comma_quibbling

I still don't understand, these challenges are supposed to be taken from those rosetta challenges. Is it cause of license/copywriting? So you're thinking that we can't copy it directly like that because of licensing? That may be the case, I'm not sure.

**Comment by Sboonny** [comment]
They are under https://www.gnu.org/licenses/old-licenses/fdl-1.2.html

Got it from https://rosettacode.org/wiki/Rosetta_Code:Privacy_policy 

So I AM worried for nothing I revert back them? Should I keep the variables names?

**Comment by moT01** [comment]
I would have preferred to keep the original, as it aligns with what's on the rosetta site.

File: curriculum/challenges/english/10-coding-interview-prep/rosetta-code/comma-quibbling.md [hunk 3]
```diff
@@ -8,29 +8,18 @@ dashedName: comma-quibbling
...
-  <li>["ABC", "DEF", "G", "H"]</li>
-</ul>
-
-**Note:** Assume words are non-empty strings of uppercase characters for this task.
+|             Array          |           Output         |
+| -------------------------- | ------------------------ |
+| []                         | "{}"                     |
+| ["ABC"]                    | "{ABC}"                  |
+| ["ABC", "DEF"]             | "{ABC and DEF}"          |
+| ["ABC", "DEF", "G", "H"]   | "{ABC, DEF, G and H}"    |
```
Comment (moT01) [inline curriculum/challenges/english/10-coding-interview-prep/rosetta-code/comma-quibbling.md:22]:
```suggestion
| []                         | {}                       |
| ["ABC"]                    | {ABC}                    |
| ["ABC", "DEF"]             | {ABC and DEF}            |
| ["ABC", "DEF", "G", "H"]   | {ABC, DEF, G and H}      |
```

Pretty minor, but the quotes didn't seem like the belonged. What do you think @Sboonny?

**Review by moT01 — COMMENTED** [review]
I still don't get why we are changing this up. The issue is to remove external links - I would have only changed line 11, the rest should have stayed as it was in my opinion. Some other reviewers didn't seem to have a problem with it so I am going to defer to them to finish reviewing this. Maybe it's no big deal - the task is still the same, I actually think the table helps, but that's not how rosetta worded it - so I would have left it.

**Comment by Sboonny** [comment]
> The issue is to remove external links - I would have only changed line 11

You are right, I will open another issue later with better alternative.

> I still don't get why we are changing this up

I had difficultiy understanding it, and it didn't feel fair because it wasn't the logic that I didn't understand but the wording itself.

> Some other reviewers didn't seem to have a problem with it so I am going to defer to them to finish reviewing this. 

there is no reason to defer, if I have issue with the wording and how it's unfair for the logic, I should open new issue. instead of taking people time and attention from the original issue.

File: curriculum/challenges/english/10-coding-interview-prep/rosetta-code/comma-quibbling.md [hunk 4]
```diff
@@ -52,19 +52,19 @@ assert(typeof quibble(['ABC']) === 'string');
...
 
-`quibble(["ABC", "DEF"])` should return "{ABC and DEF}".
+`quibble(["ABC", "DEF"])` should return `"{ABC and DEF}"`.
 
 ```js
 assert.equal(quibble(testCases[2]), results[2]);
 ```
 
-`quibble(["ABC", "DEF", "G", "H"])` should return "{ABC,DEF,G and H}".
+`quibble(["ABC", "DEF", "G", "H"])` should return `"{ABC, DEF, G and H}"`.
```
Comment (moT01) [inline curriculum/challenges/english/10-coding-interview-prep/rosetta-code/comma-quibbling.md:67]: Sorry to drag this out even further - but this pointed out something to me, the test on line 79 is looking for `"{ABC,DEF,G and H}"` - if the function returns it with the spaces, it fails:
Screen Shot 2022-06-16 at 10 32 07 AM
 
The instructions say that spaces should be added. Do you mind adjusting that test real quick?

**Review by moT01 — COMMENTED** [review]
> Goodbye table, I fought hard for you

Lol, sorry @Sboonny. It really is just my opinion and preference that we keep the wording aligned with what's on the rosetta site if possible. Again, others didn't seem to have a problem with it, so we could continue the discussion.

I found one last issue.

**Review by moT01 — APPROVED** [review]
LGTM 🎉 Want to give this a quick look @ShaunSHamilton

**Review by Sembauke — APPROVED** [review]

**Pull request merged**